### PR TITLE
Fix coordination test 2.13.0-M5 #26636

### PR DESCRIPTION
**Pull request by raboof**
Workaround for https://github.com/scala/bug/issues/11242

Refs #26636

### CODE REVIEW

File: akka-actor-tests/src/test/scala/akka/actor/DynamicAccessSpec.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        case f: Failure[_] => f
+      }
+
+  }
+
+  override def afterAll() = {
+    Await.result(system.terminate(), 10.seconds)
+    super.afterAll()
+  }
+}
```
Comment (johanandren) [inline akka-actor-tests/src/test/scala/akka/actor/DynamicAccessSpec.scala]: Was there no test coverage before?

File: akka-coordination/src/main/scala/akka/coordination/lease/scaladsl/LeaseProvider.scala [hunk 2]
```diff
@@ -65,15 +64,17 @@ class LeaseProvider(system: ExtendedActorSystem) extends Extension {
...
+    val instance: Try[Lease] = dynamicAccess.createInstanceFor[Lease](
+      fqcn,
+      immutable.Seq((classOf[LeaseSettings], leaseSettings), (classOf[ExtendedActorSystem], system))) match {
+      case s: Success[Lease] =>
+        s
+      case Failure(_: NoSuchMethodException) =>
+        dynamicAccess.createInstanceFor[Lease](fqcn, immutable.Seq((classOf[LeaseSettings], leaseSettings)))
+      case f: Failure[_] =>
+        f
+    }
```
Comment (johanandren) [inline akka-coordination/src/main/scala/akka/coordination/lease/scaladsl/LeaseProvider.scala]: This is the actual fix, match instead of `recoverWith`? 👍

**Review by johanandren — APPROVED** [review]

File: akka-actor-tests/src/test/scala/akka/actor/DynamicAccessSpec.scala [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  def name: String
+}
+
+class TestClassWithStringConstructor(val name: String) extends TestSuperclass
+class TestClassWithDefaultConstructor extends TestSuperclass {
+  override def name = "default"
+}
+
+class DynamicAccessSpec extends WordSpec with Matchers with BeforeAndAfterAll {
+  val system = ActorSystem()
```
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/actor/DynamicAccessSpec.scala]: why not use AkkaSpec?

File: akka-coordination/src/main/scala/akka/coordination/lease/scaladsl/LeaseProvider.scala [hunk 4]
```diff
@@ -65,15 +64,17 @@ class LeaseProvider(system: ExtendedActorSystem) extends Extension {
...
-          dynamicAccess.createInstanceFor[Lease](fqcn, immutable.Seq((classOf[LeaseSettings], leaseSettings)))
-
-      } match {
+    val instance: Try[Lease] = dynamicAccess.createInstanceFor[Lease](
+      fqcn,
+      immutable.Seq((classOf[LeaseSettings], leaseSettings), (classOf[ExtendedActorSystem], system))) match {
+      case s: Success[Lease] =>
+        s
+      case Failure(_: NoSuchMethodException) =>
+        dynamicAccess.createInstanceFor[Lease](fqcn, immutable.Seq((classOf[LeaseSettings], leaseSettings)))
```
Comment (patriknw) [inline akka-coordination/src/main/scala/akka/coordination/lease/scaladsl/LeaseProvider.scala]: We have another place referencing this bug, but it's still using recoverWith
```
    system.dynamicAccess.createInstanceFor[Serializer](fqn, List(classOf[ExtendedActorSystem] -> system)).recoverWith {
      case _: NoSuchMethodException =>
        system.dynamicAccess.createInstanceFor[Serializer](fqn, Nil)
      // FIXME only needed on 2.13.0-M5 due to https://github.com/scala/bug/issues/11242
      case t => Failure(t)
    }
```

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-actor-tests/src/test/scala/akka/actor/DynamicAccessSpec.scala (hunk 1 above)
Comment (raboof) [reply akka-actor-tests/src/test/scala/akka/actor/DynamicAccessSpec.scala]: Not explicitly at least

File: akka-coordination/src/main/scala/akka/coordination/lease/scaladsl/LeaseProvider.scala (hunk 2 above)
Comment (raboof) [reply akka-coordination/src/main/scala/akka/coordination/lease/scaladsl/LeaseProvider.scala]: yes, fix/workaround ;)

**Pull request merged**